tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -98,6 +98,84 @@ struct DotOpConverter : public OpRewritePattern<DotOp> {
   }
 };
 
+/// Concat Operation Example (2D):
+/// Given inpA[2][1], inpB[2][2], concat_dimension = 1.
+/// Compute output[x1][x2].
+/// Implementation Pseudocode:
+/// s = 0
+/// for a in range(0, 2):
+///   for b in range(0, 1):
+///     output[a][b] = inpA[a][b - s]
+/// s = 1
+/// for a in range(0, 2):
+///   for b in range(1, 3):
+///     output[a][b] = inpB[a][b - s]
+///
+/// Concatenate composes an array from multiple array operands. The array is of
+/// the same rank as each of the input array operands (which must be of the same
+/// rank as each other) and contains the arguments in the order that they were
+/// specified.
+struct ConcatOpConverter : public OpRewritePattern<ConcatenateOp> {
+  using OpRewritePattern<ConcatenateOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ConcatenateOp op,
+                                PatternRewriter& rewriter) const override {
+    Location loc = op.getLoc();
+    Value output = op.output();
+    MemRefType outputType = output.getType().cast<MemRefType>();
+    unsigned outputRank = outputType.getRank();
+    ArrayRef<int64_t> outputShape = outputType.getShape();
+
+    ValueRange operands = op.val();
+    uint64_t concatDim = op.dimension();
+    int64_t prevBound = 0;
+
+    for (Value operand : operands) {
+      MemRefType operandType = operand.getType().cast<MemRefType>();
+      ArrayRef<int64_t> operandShape = operandType.getShape();
+
+      // TODO: Extend support for dynamic dimensions.
```

Comment:
Our system blocks the commit because `TODO` must be spelled `TODO(username)` ... Can you update here and below?

Reply:
@joker-eph Done.